[bug:1193929] GlusterFS can be improved

URL: https://bugzilla.redhat.com/1193929
Creator: jdarcy at redhat
Time: 20150218T15:06:47

I hope this bug is never fixed.

The upstream patch process requires that each patch have an associated bug ID before it can be merged.  However, there is no requirement that the bug contain any information or receive any kind of signoff before the patch can proceed.  As a result, many of our developers have the habit of creating such "placeholder" bugs every time they want to make a change, even if it's just a random cleanup/idea and not an actual bug fix or requested/tracked feature request.  Also, any patch with the dreaded "rfc" bug ID (which would be appropriate for such changes) is unlikely to be reviewed.

This bug exists to satisfy our process requirement, without the additional negatives of clogging up our triage/tracking processes and making it appear that the code has more bugs (which would be bad) when in fact a developer had more ideas (which is generally good).

Time: 20180820T03:03:00
bugzilla-bot at gluster.org commented: 
COMMIT: https://review.gluster.org/20746 committed in master by "Amar Tumballi" <amarts@redhat.com> with a commit message- gnfs: fix gcc warnings

gcc version 8.1.1 20180712 (Red Hat 8.1.1-5) (GCC) on Fedora 28.

Sample warnings:
1)
nfs3.c:292:54: warning: ‘%s’ directive output may be truncated writing up to 255 bytes into a region of size between 143 and 250 [-Wformat-truncation=]
                         snprintf (buf, sizeof (buf), "(%s) %s : %s",    \
                                                      ^~~~~~~~~~~~~~
nfs3.c:295:35:
                                   gfid);                                \
                                   ~~~~
2)
nlm4.c:145:54: warning: ‘%s’ directive output may be truncated writing up to 255 bytes into a region of size between 143 and 250 [-Wformat-truncation=]
                         snprintf (buf, sizeof (buf), "(%s) %s : %s",    \
                                                      ^~~~~~~~~~~~~~
nlm4.c:148:35:
                                   gfid);                                \
                                   ~~~~
acl3.c:128:54: warning: ‘%s’ directive output may be truncated writing up to 255 bytes into a region of size between 143 and 250 [-Wformat-truncation=]
                         snprintf (buf, sizeof (buf), "(%s) %s : %s",    \
                                                      ^~~~~~~~~~~~~~
acl3.c:131:35:
                                   gfid);                                \
                                   ~~~~

updates: bz#1193929
Change-Id: I760b8176e48f1f4628a1a98afa54a7994bdf13e9
Signed-off-by: Ravishankar N <ravishankar@redhat.com>

Time: 20180821T10:50:59
bugzilla-bot at gluster.org commented: 
COMMIT: https://review.gluster.org/20744 committed in master by "Amar Tumballi" <amarts@redhat.com> with a commit message- glusterd: fix gcc warnings

...and also changed char array initialization to from {0,} to "".

gcc version 8.1.1 20180712 (Red Hat 8.1.1-5) (GCC) on Fedora 28.

Sample warnings:
glusterd-utils.c:7234:41: warning: ‘.hostname’ directive output may be truncated writing 9 bytes into a region of size between 1 and 1024 [-Wformat-truncation=]
         snprintf (key, sizeof (key), "%s.hostname", base_key);
                                         ^~~~~~~~~
glusterd-utils.c:7234:9: note: ‘snprintf’ output between 10 and 1033 bytes into a destination of size 1024
         snprintf (key, sizeof (key), "%s.hostname", base_key);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

glusterd-snapshot.c:3090:65: warning: ‘/’ directive output may be truncated writing 1 byte into a region of size between 0 and 4095 [-Wformat-truncation=]
                 snprintf (snap_path, sizeof (snap_path) - 1, "%s/%s",
                                                                 ^
glusterd-snapshot.c:3090:17: note: ‘snprintf’ output between 2 and 4351 bytes into a destination of size 4095
                 snprintf (snap_path, sizeof (snap_path) - 1, "%s/%s",
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                           snap_mount_dir, snap_vol->snapshot->snapname);
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

glusterd-statedump.c:28:45: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size 144 [-Wformat-truncation=]
         snprintf (subkey, sizeof (subkey), "%s%d", key, index);
                                             ^~     ~~~
glusterd-statedump.c:28:9: note: ‘snprintf’ output between 2 and 4107 bytes into a destination of size 144
         snprintf (subkey, sizeof (subkey), "%s%d", key, index);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

updates: bz#1193929
Change-Id: Ic721f27b28d1221c124b570e81c55528f5b7f3cd
Signed-off-by: Ravishankar N <ravishankar@redhat.com>

--- a/extras/ganesha/scripts/ganesha-ha.sh
+++ b/extras/ganesha/scripts/ganesha-ha.sh
@@ -948,18 +948,18 @@ status()
     # check if the VIP and port block/unblock RAs are on the expected nodes
     for n in ${nodes[*]}; do
 
-        grep -E -x "${n}-nfs_block \(ocf::heartbeat:portblock\): Started ${n}" > /dev/null 2>&1 ${scratch}
+        grep -E -x "${n}-nfs_block +\(ocf::heartbeat:portblock\): +Started ${n}" > /dev/null 2>&1 ${scratch}
         result=$?
         ((healthy+=${result}))
-        grep -E -x "${n}-cluster_ip-1 \(ocf::heartbeat:IPaddr\): Started ${n}" > /dev/null 2>&1 ${scratch}
+        grep -E -x "${n}-cluster_ip-1 +\(ocf::heartbeat:IPaddr\): +Started ${n}" > /dev/null 2>&1 ${scratch}
         result=$?
         ((healthy+=${result}))
-        grep -E -x "${n}-nfs_unblock \(ocf::heartbeat:portblock\): Started ${n}" > /dev/null 2>&1 ${scratch}
+        grep -E -x "${n}-nfs_unblock +\(ocf::heartbeat:portblock\): +Started ${n}" > /dev/null 2>&1 ${scratch}
         result=$?
         ((healthy+=${result}))
     done
 
-    grep -E "\):\ Stopped|FAILED" > /dev/null 2>&1 ${scratch}
+    grep -E "\): +Stopped|FAILED" > /dev/null 2>&1 ${scratch}
     result=$?
 
     if [ ${result} -eq 0 ]; then

